File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int l = 0, r = 1;
        int res = 0;
        while(r < nums.size()){
            if(nums[l] == nums[r]){
                res = nums[l];
                break;
            }else{
                r++;
                if(r == nums.size()){
                    l++;
                    r= l+1;
                }
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/burst-balloons/submission-0.py
class Solution:
    def maxCoins(self, nums: List[int]) -> int:
        nums = [1] + nums + [1]
        memo = {}

        def dfs(left, right):
            if (left, right) in memo:
                return memo[(left, right)]

            if left + 1 == right:
                return 0
            

            res = 0
            for k in range(left + 1, right):
                coins = dfs(left, k) + dfs(k, right) + nums[left] * nums[k] * nums[right]
                res = max(res, coins)

            memo[(left, right)] = res
            return res

        return dfs(0, len(nums) - 1)